src/parser/jobs/pld/modules/Gauge.tsx

```diff
@@ -1,55 +1,57 @@
 import {t} from '@lingui/macro'
 import {Trans} from '@lingui/react'
-import {ActionLink} from 'components/ui/DbLink'
-import ACTIONS from 'data/ACTIONS'
+import {DataLink} from 'components/ui/DbLink'
 import JOBS from 'data/JOBS'
-import {CastEvent} from 'fflogs'
+import {Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
 import {dependency} from 'parser/core/Module'
-import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import {Data} from 'parser/core/modules/Data'
+import {CounterGauge, CounterGaugeModifier, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
 import Suggestions, {SEVERITY, Suggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 
-interface GaugeModifier {
-	[key: string]: number | undefined
-}
-
-const OATH_MODIFIERS = new Map<number, GaugeModifier>([
-	[ACTIONS.ATTACK.id, {cast: 5}],
-	[ACTIONS.SHELTRON.id, {cast: -50}],
-	[ACTIONS.INTERVENTION.id, {cast: -50}],
-	[ACTIONS.COVER.id, {cast: -50}],
-])
 export default class Gauge extends CoreGauge {
 	static override title = t('pld.gauge.title')`Oath Gauge Usage`
 
+	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private oathGauge = this.add(new CounterGauge({
 		chart: {label: 'Oath Gauge', color: JOBS.PALADIN.colour},
 	}))
-
-	protected override init() {
-		super.init()
-
-		this.addEventHook('cast', {by: 'player', abilityId: Array.from(OATH_MODIFIERS.keys())}, this.onOathModifying)
+	private oathModifiers = new Map<number, CounterGaugeModifier>([
```

Comment:
nts; seeing a bunch of this cropping up. i should probably set up some helpers to cover common data key mapping tasks on module init